File: 2025-03-March-GFG-POTD/March 08 - Longest Palindrome in a String/Solution.cpp
#include <string>
using namespace std;

class Solution {
public:
    string longestPalindrome(string s) {
        if (s.empty()) return "";
        int start = 0, max_len = 0;
        
        for (int i = 0; i < s.size(); ++i) {
            int len1 = expand(s, i, i);      // Odd-length palindromes
            int len2 = expand(s, i, i + 1);  // Even-length palindromes
            int current_len = max(len1, len2);
            
            if (current_len > max_len) {
                max_len = current_len;
                // Calculate start index based on expansion type
                start = (len1 > len2) ? (i - (len1 - 1)/2) : (i - (len2/2 - 1));
            }
        }
        return s.substr(start, max_len);
    }
    
private:
    int expand(const string& s, int left, int right) {
        while (left >= 0 && right < s.size() && s[left] == s[right]) {
            left--;
            right++;
        }
        return right - left - 1; // Actual palindrome length
    }
};

File: 2025-03-March-GFG-POTD/March 08 - Longest Palindrome in a String/Solution.py
def longestPalindrome(s):
    if not s:
        return ""
    
    start = 0
    max_len = 0
    
    def expand(left, right):
        while left >= 0 and right < len(s) and s[left] == s[right]:
            left -= 1
            right += 1
        return right - left - 1  # Actual palindrome length
    
    for i in range(len(s)):
        len1 = expand(i, i)      # Odd-length palindromes
        len2 = expand(i, i + 1)  # Even-length palindromes
        current_len = max(len1, len2)
        
        if current_len > max_len:
            max_len = current_len
            # Calculate start index based on expansion type
            if len1 > len2:
                start = i - (len1 - 1) // 2
            else:
                start = i - (len2 // 2 - 1)
    
    return s[start:start + max_len]
